editor/store/effects.js

```diff
@@ -179,6 +186,37 @@ export default {
 			__( 'Updating failed' );
 		dispatch( createErrorNotice( noticeMessage, { id: SAVE_POST_NOTICE_ID } ) );
 	},
+	REQUEST_AUTOSAVE_EXISTS( action, store ) {
+		const { autosaveStatus } = action;
+		const { dispatch } = store;
+		const noticeMessage = __( 'There is an autosave of this post that is more recent than the version below.' );
+		if ( autosaveStatus ) {
+			dispatch( createWarningNotice(
+				<p>
+					<span>{ noticeMessage }</span>
+					{ ' ' }
+					{ <a href={ autosaveStatus.edit_link }>{ __( 'View the autosave' ) }</a> }
+				</p>,
+				{
+					id: AUTOSAVE_POST_NOTICE_ID,
+					spokenMessage: noticeMessage,
+				}
+			) );
+		}
+	},
+	REQUEST_CONNECTION_LOST( action, store ) {
+		const { dispatch } = store;
+		const noticeMessage = __( 'You have lost your connection with the server, and saving has been disabled. This message will vanish once you\'ve reconnected.' );
+		dispatch( createWarningNotice(
+			<p>
```

Comment:
Does a notice message need to have `<p>` or `<span>` wrappers? Ideally it shouldn't.